### DISCUSSION TITLE
CLR namespace whitelist

### DISCUSSION THREAD

**Discussion by philpax** [sebastienros/jint#1133] [Q&A] [answered]
Hi there!

Love the work you've done with jint. I'm looking at embedding it into a C# game as a simple untrusted user scripting solution, and I was wondering if there was a way to enable CLR interop while minimising what's exposed to the script.

Ideally, I'm looking for something that will let me enforce a whitelist like this: https://wiki.facepunch.com/sbox/AccessList#accesslist

I had a look through the repo and found [this comment](https://github.com/sebastienros/jint/issues/222#issuecomment-170644077), which I think should work, but I'd love to use a more granular mechanism if it exists.

Thanks in advance!

**Comment by lahma** [comment, marked as answer]
I think you should be able to do all the filtering you need by combining `AllowClr` call with wanted assemblies + using TypeResolver, [here are some test cases](https://github.com/sebastienros/jint/blob/main/Jint.Tests/Runtime/InteropTests.MemberAccess.cs) covering member + types,

**Reply by philpax** [reply]
Fantastic, thank you! I'll have a go and mark this as the answer once I can test it 🙂
